### PR TITLE
"TreeNodeModelAdminInline"

### PR DESCRIPTION
**Pull request by domlysi**
Inline Mixin to edit children in their parent and added default inline if no inlines are defined.

Either use the TreeNodeModelAdminInline as a MixIn for customizing the inline or just leave it with the default.

### CODE REVIEW

File: treenode/admin.py [hunk 1]
```diff
@@ -6,6 +6,17 @@
 from .forms import TreeNodeForm
 
 
+class TreeNodeModelAdminInline(admin.TabularInline):
+    model, instance = None, None
+
+    def get_formset(self, request, obj=None, **kwargs):
+        self.instance = obj
+        return super().get_formset(request, obj, **kwargs)
```
Comment (fabiocaccamo) [inline treenode/admin.py:14]: This package supports python 2.7, so the super call must receive the 2 arguments.

File: treenode/admin.py [hunk 2]
```diff
@@ -38,6 +49,15 @@ class MyModelAdmin(TreeNodeModelAdmin):
     list_per_page = 1000
     ordering = ('tn_order', )
 
+    def get_inlines(self, request, obj):
+        inlines = super().get_inlines(request, obj)
```
Comment (fabiocaccamo) [inline treenode/admin.py:53]: This package supports python 2.7, so the super call must receive the 2 arguments.

File: treenode/admin.py [hunk 3]
```diff
@@ -56,8 +76,6 @@ def treenode_field_display(obj):
                 base_list_display.pop(0)
             return (treenode_field_display, ) + tuple(base_list_display)
 
-        return base_list_display
-
```
Comment (domlysi) [inline treenode/admin.py:60]: if and else both return so this one is unnecessary

**Comment by fabiocaccamo** [comment]
@domlysi I don't understand the point of adding the inline by default, could you explain me it better?

**Comment by domlysi** [comment]
As far as I remember we used it to allow to edit the children of the particular element straight in the admin interface.

**Comment by fabiocaccamo** [comment]
I also use to do it, but adding a standard inline is enough, I don't see the plus of integrating the abstract inline inside the library.

**Comment by domlysi** [comment]
Because we got this feature request by our customer  and to improve UX in the admin system, we decided to keep it as standard in our projects. But there is no need, it's just for convenience. I thought it could be useful to other too.

**Pull request closed without merging**